keys/keys.go

```diff
@@ -357,3 +357,17 @@ func MetaScanBounds(key proto.Key) (proto.Key, proto.Key) {
 	// Otherwise find the first entry greater than the given key in the same meta prefix.
 	return key.Next(), proto.Key(key[:len(Meta1Prefix)]).PrefixEnd()
 }
+
+// MetaReverseScanBounds returns the start and end keys of the range within which the
+// desired meta record can be found by means of an engine reverse scan. The given key
+// must be a valid RangeMetaKey as defined by ValidateRangeMetaKey.
+func MetaReverseScanBounds(key proto.Key) (proto.Key, proto.Key, error) {
+	if key.Equal(proto.KeyMin) || key.Equal(Meta1Prefix) {
+		return nil, nil, NewInvalidRangeMetaKeyError("KeyMin or Meta1Prefix can't be used as the key of reverse scan", key)
+	}
+	if key.Equal(Meta2Prefix) {
+		return Meta1Prefix, Meta1Prefix.PrefixEnd(), nil
+	}
+	// Otherwise find the first entry greater than the given key in the same meta prefix.
+	return key[:len(Meta1Prefix)], key.Next(), nil
```

Comment:
In this scenario, if we drop the next(), "f" is set to the endkey of the `MvccReverseScan`  and the end key is exclusive.